src/crimson/net/ProtocolV1.cc

```diff
@@ -826,8 +826,10 @@ seastar::future<> ProtocolV1::read_message()
     }).then([this] (bufferlist bl) {
       auto p = bl.cbegin();
       ::decode(m.footer, p);
+      auto pconn = seastar::static_pointer_cast<SocketConnection>(
+        conn.shared_from_this());
       auto msg = ::decode_message(nullptr, 0, m.header, m.footer,
-                                  m.front, m.middle, m.data, nullptr);
+                                  m.front, m.middle, m.data, std::move(pconn));
```

Comment:
> Message has non-null ref to SocketConnection now.

could you elaborate a little bit the rationale behind this change?

Reply:
It's enabler for some functions from the Object Class API and the watch/notify mechanism:

* `cls_get_client_features()`,
* `cls_get_request_origin()`.
* e.g. `CEPH_OSD_OP_NOTIFY`, `CEPH_OSD_OP_NOTIFY_ACK` and `Notify::maybe_complete_notify()`.

-----------
I'm going to put this into the commit message.